test/e2e/ingress_test.go

```diff
@@ -1049,5 +1053,204 @@ var _ = Describe("Ingress", func() {
+												ServiceName: echoSvc1.Name,
+												ServicePort: intstr.FromInt(int(echoSvc1.Spec.Ports[0].Port)),
+											},
+										},
+									},
+								},
+							},
+						},
+					}
+				})
+				Expect(err).NotTo(HaveOccurred(), "failed to create ingress")
+
+				// Wait for EdgeLB to acknowledge the EdgeLB pool's creation.
+				err = retry.WithTimeout(framework.DefaultRetryTimeout, framework.DefaultRetryInterval, func() (bool, error) {
+					ctx, fn := context.WithTimeout(context.Background(), framework.DefaultRetryInterval/2)
+					defer fn()
+					pool, err = f.EdgeLBManager.GetPool(ctx, *httpEchoSpec.Name)
+					return err == nil, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "timed out while waiting for the edgelb api server to acknowledge the pool's creation")
+
+				// Make sure the pool is reporting the requested configuration.
+				Expect(pool.Name).To(Equal(*httpEchoSpec.Name))
+
+				// Wait for the Ingress to be reachable.
+				log.Debugf("waiting for the public ip for %q to be reported", kubernetes.Key(ingress))
+				err = retry.WithTimeout(framework.DefaultRetryTimeout, framework.DefaultRetryInterval, func() (bool, error) {
+					// Wait for the pool's public IP to be reported.
+					ctx, cancel := context.WithTimeout(context.Background(), framework.DefaultRetryTimeout)
+					defer cancel()
+					publicIP, err = f.WaitForPublicIPForIngress(ctx, ingress)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(publicIP).NotTo(BeEmpty())
+
+					// Attempt to connect to the host using the reported IP.
+					addr := fmt.Sprintf("http://%s:%d", host, *httpEchoSpec.Frontends.HTTPS.Port)
+					log.Debugf("attempting to connect to %q at %q via %v", kubernetes.Key(ingress), addr, publicIP)
+
+					caCertPool := x509.NewCertPool()
+					caCertPool.AppendCertsFromPEM(crt)
+
+					f.HTTPClient.Transport = &http.Transport{
+						DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+							// Similar to curls --resolve behavior. Use a custom
+							// dialer to force resolve the host (ex.
+							// foo.bar.com) to the reported IP.
+							addr = fmt.Sprintf("%s:%d", publicIP, *httpEchoSpec.Frontends.HTTPS.Port)
+							tlsConfig := &tls.Config{
+								RootCAs:    caCertPool,
+								ServerName: "foo.bar.com",
+							}
+							return tls.Dial("tcp", addr, tlsConfig)
+						},
+					}
+					defer func() {
+						// reset TLS configuration
+						f.HTTPClient.Transport = &http.Transport{}
+					}()
+
+					r, err := f.HTTPClient.Get(addr)
+					if err != nil {
+						log.Debugf("waiting for the ingress to be reachable at %q", addr)
+						return false, nil
+					}
+					log.Debugf("the ingress is reachable at %q", addr)
+					return r.StatusCode == 200, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), "timed out while waiting for the ingress to be reachable")
+
+				// Manually delete the Ingress resource now so that the target EdgeLB pool isn't left dangling after namespace deletion.
```

Comment:
Shouldn't we do this in `defer` and perhaps wait until the edge-lb pool is deleted?

Reply:
Done.